src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -7347,15 +7347,22 @@ private StatementSyntax ParseEmbeddedStatement(bool complexCheck)
                 statement = SyntaxFactory.EmptyStatement(EatToken(SyntaxKind.SemicolonToken));
             }
 
-            // An "embedded" statement is simply a statement that is not a labelled
-            // statement or a declaration statement.
             switch (statement.Kind)
             {
+                // An "embedded" statement is simply a statement that is not a labelled
+                // statement or a declaration statement.
                 case SyntaxKind.LabeledStatement:
                 case SyntaxKind.LocalDeclarationStatement:
                 case SyntaxKind.LocalFunctionStatement:
                     statement = this.AddError(statement, ErrorCode.ERR_BadEmbeddedStmt);
                     break;
+                // Expression statements as embedded statements should be followed by semicolon.
```

Comment:
The comment is incorrect in a very confusing way. An expression statement is a statement-expression followed by a semicolon. It doesn't matter if it is embedded or not. An expression statement *always* has a semicolon as part of its syntax.